verify.go

```diff
@@ -8,21 +8,42 @@ import (
 	"github.com/mastahyeti/cms/protocol"
 )
 
+func getLeafs(chains [][][]*x509.Certificate, inputErr error) (leafs []*x509.Certificate, err error) {
+	if err = inputErr; err != nil {
+		return
+	}
+	leafs = make([]*x509.Certificate, len(chains))
+	for i, chain := range chains {
+		leafs[i] = chain[i][0]
+	}
+	return
+}
+
+
 // Verify verifies the SingerInfos' signatures. Each signature's associated
 // certificate is verified using the provided roots. UnsafeNoVerify may be
 // specified to skip this verification. Nil may be provided to use system roots.
 // The certificates whose keys made the signatures are returned.
 //
 // WARNING: this function doesn't do any revocation checking.
 func (sd *SignedData) Verify(opts x509.VerifyOptions) ([]*x509.Certificate, error) {
+	return getLeafs(sd.VerifyFull(opts))
+}
+
+// VerifyFull does the same process than Verify, but it returns the full certificate chains used
+// to verify each of the signatures. This way you can use this information in case you want to 
+// validate the revocation status of the certificates in the chain
+//
+// WARNING: As Verify, this function does not do any revocation checking
+func (sd *SignedData) VerifyFull(opts x509.VerifyOptions) ([][][]*x509.Certificate, error) {
 	econtent, err := sd.psd.EncapContentInfo.EContentValue()
 	if err != nil {
 		return nil, err
 	}
 	if econtent == nil {
 		return nil, errors.New("detached signature")
 	}
-
+	
```

Comment:
There are a few whitespace issues in this diff. Please run `go fmt` to fix them.